docs/installation/sources/sources_mac.rst

```diff
@@ -0,0 +1,370 @@
+This section explains how to use it to compile *eProsima Fast DDS* and its dependencies.
+
+#. Install the ROS 2 development tools (colcon_ and vcstool_) by executing the following command:
+
+   .. code-block:: bash
+
+       pip3 install -U colcon-common-extensions vcstool
+
+
+#. Create a :code:`Fast-DDS` directory and download the repos file that will be used to install
+   *eProsima Fast DDS* and its dependencies:
+
+   .. code-block:: bash
+
+       mkdir ~/Fast-DDS
+       cd ~/Fast-DDS
+       wget https://raw.githubusercontent.com/eProsima/Fast-DDS/master/fastrtps.repos
+       mkdir src
+       vcs import src < fastrtps.repos
+
+#. Build the packages:
+
+   .. code-block:: bash
+
+       colcon build --cmake-args -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DOPENSSL_LIBRARIES=/usr/local/opt/openssl/lib
+
+.. note::
+
+    The :code:`--cmake-args` option allows to pass the CMake configuration options to the :code:`colcon build` command.
+    In Mac OS the location of OpenSSL is not found automatically and therefore has to be passed explicitly.
+    For more information on the specific syntax, please refer to the `CMake specific arguments
+    <https://colcon.readthedocs.io/en/released/reference/verb/build.html#cmake-specific-arguments>`_ page of the colcon_
+    manual.
+
+.. _run_app_colcon_sm:
+
+Run an application
+^^^^^^^^^^^^^^^^^^
+
+When running an instance of an application using *eProsima Fast DDS*, the colcon overlay built in the
+dedicated :code:`Fast-DDS` directory must be sourced.
+There are two possibilities:
+
+* Every time a new shell is opened, prepare the environment locally by typing the
+  command:
+
+  .. code-block:: bash
+
+      source ~/Fast-DDS/install/setup.bash
+
+* Add the sourcing of the colcon overlay permanently to the :code:`PATH`, by typing the following:
+
+  .. code-block:: bash
+
+      touch ~/.bash_profile
+      echo 'source ~/Fast-DDS/install/setup.bash' >> ~/.bash_profile
+
+
+.. _cmake_installation_mac:
+
+CMake installation
+------------------
+
+This section explains how to compile *eProsima Fast DDS* with CMake_, either :ref:`locally <local_installation_sm>` or
+:ref:`globally <global_installation_sm>`.
+
+.. _local_installation_sm:
+
+Local installation
+^^^^^^^^^^^^^^^^^^
+
+#. Create a :code:`Fast-DDS` directory where to download and build *eProsima Fast DDS* and its dependencies:
+
+   .. code-block:: bash
+
+       mkdir ~/Fast-DDS
+
+#. Clone the following dependencies and compile them using CMake_.
+
+   * `Foonathan memory <https://github.com/foonathan/memory>`_
+
+     .. code-block:: bash
+
+         cd ~/Fast-DDS
+         git clone https://github.com/eProsima/foonathan_memory_vendor.git
+         mkdir foonathan_memory_vendor/build
+         cd foonathan_memory_vendor/build
+         cmake .. -DCMAKE_INSTALL_PREFIX=~/Fast-DDS/install -DBUILD_SHARED_LIBS=ON -DOPENSSL_ROOT_DIR=/usr/local/opt/openssl -DOPENSSL_LIBRARIES=/usr/local/opt/openssl/lib
```

Comment:
```suggestion
         cmake .. -DCMAKE_INSTALL_PREFIX=~/Fast-DDS/install -DBUILD_SHARED_LIBS=ON
```
OpenSSL is not needed for foonathan_memory. It is only required by Fast DDS and only when SECURITY is enabled.